route_generator/src/route_generator.cpp

```diff
@@ -0,0 +1,118 @@
+/*
+ * Copyright (C) 2019 LEIDOS.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License. You may obtain a copy of
+ * the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+
+#include <ros/ros.h>
+
+#include "route_generator.h"
+
+RouteGenerator::RouteGenerator(int argc, char **argv, const std::string &name) : route_is_active_(false)
+{
+    ros::init(argc, argv, name);
+}
+
+RouteGenerator::~RouteGenerator() {}
+
+void RouteGenerator::initialize()
+{
+    nh_.reset(new ros::CARMANodeHandle);
+    pnh_.reset(new ros::CARMANodeHandle("~"));
+    nh_->getParam("route_file_path", route_file_path_);
```

Comment:
you should use the pnh_ here as this is a parameter

Reply:
Good point.